diff --git a/include/otawa/ai/EdgeStore.h b/include/otawa/ai/EdgeStore.h
index 00678a70..5b1ea8ae 100644
--- a/include/otawa/ai/EdgeStore.h
+++ b/include/otawa/ai/EdgeStore.h
@@ -44,16 +44,18 @@ public:
 	}
 
 	void set(vertex_t v, t s) {
-		for(auto e =_graph.succs(v); e; e++)
-			map.put(*e, s);
+		for(typename G::Successor e(_graph, v); e(); e++)
+                        map.put(*e, s);
+
 	}
 
 	inline void set(edge_t e, const t& s) { map.put(e, s); }
 
 	t get(vertex_t v) const {
 		t s = _dom.bot();
-		for(auto e =_graph.succs(v); e; e++)
-			_dom.join(s, map.get(e, _dom.bot()));
+		for(typename G::Successor e(_graph, v); e(); e++)
+                        _dom.join(s, map.get(e, _dom.bot()));
+
 		return s;
 	}
 
diff --git a/include/otawa/ai/WorkListDriver.h b/include/otawa/ai/WorkListDriver.h
index 66bee111..2df79da0 100644
--- a/include/otawa/ai/WorkListDriver.h
+++ b/include/otawa/ai/WorkListDriver.h
@@ -236,7 +236,7 @@ public:
 	OrderedDriver(D& dom, const G& graph, S& store, O *order)
 	: _dom(dom), _graph(graph), _store(store), wl_set(graph.count()), end(false), _order(order) {
 		store.set(_graph.entry(), dom.init());
-		for(typename G::Successor succ(graph, _graph.entry()); succ; succ++)
+		for(typename G::Successor succ(graph, _graph.entry()); succ(); succ++)
 			push(graph.sinkOf(*succ));
 		next();
 	}
@@ -274,7 +274,7 @@ public:
 	 * (and successors must be updated).
 	 */
 	inline void change(void) {
-		for(typename G::Successor succ(_graph, cur); succ; succ++)
+		for(typename G::Successor succ(_graph, cur); succ(); succ++)
 			push(*succ);
 	}
 
@@ -354,7 +354,7 @@ public:
 	 */
 	inline typename D::t input(vertex_t vertex) {
 		typename D::t s = _dom.bot();
-		for(typename G::Predecessor pred(_graph, vertex); pred; pred++) {
+		for(typename G::Predecessor pred(_graph, vertex); pred(); pred++) {
 			s = _dom.join(s, _store.get(*pred));
 		}
 		return s;
diff --git a/src/oslice/LivenessChecker.cpp b/src/oslice/LivenessChecker.cpp
index 04dfd02b..c0485aff 100644
--- a/src/oslice/LivenessChecker.cpp
+++ b/src/oslice/LivenessChecker.cpp
@@ -61,7 +61,7 @@ LivenessChecker::LivenessChecker(AbstractRegistration& _reg) : otawa::Processor(
 void LivenessChecker::configure(const PropList &props) {
 	Processor::configure(props);
 	_debugLevel = LIVENESS_DEBUG_LEVEL(props);
-	LIVENESS_DEBUG_LEVEL(workspace()) = _debugLevel;
+//	LIVENESS_DEBUG_LEVEL(workspace()) = _debugLevel;
 }
 
 /**
